src/main/java/teammates/common/util/Sanitizer.java

```diff
@@ -115,6 +115,29 @@ public static Text sanitizeTextField(Text rawText) {
     }
 
     /**
+     * Sanitizes the string for use as a course ID by replacing illegal
+     * characters with _.
+     * 
+     * @param str String to be sanitized for use as a course ID.
+     * @return Sanitized string, a valid course ID.
+     */
+    public static String sanitizeForCourseId(String str) {
```

Comment:
We can generalize this to take 'legalChars' as a parameter? In that case, it will no longer be coupled to CourseId.